src/blenderbim/blenderbim/bim/prop.py

```diff
@@ -147,11 +148,17 @@ class Attribute(PropertyGroup):
     is_optional: BoolProperty(name="Is Optional")
     enum_items: StringProperty(name="Value")
     enum_value: EnumProperty(items=getAttributeEnumValues, name="Value", update=updateAttributeValue)
-
+    enum_data_type: StringProperty(name="Enum Data Type")
+    
     def get_value(self):
         if self.is_null:
             return None
-        return getattr(self, str(self.get_value_name()), None)
+        if self.data_type == "enum":
+            type_map = blenderbim.bim.schema.ifc.type_map
+            type_fn = {'integer':int, 'string':str, 'float':float, 'bool': bool}[type_map[self.enum_data_type]]
```

Comment:
This is ugly but not unprecedented https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/blenderbim/blenderbim/bim/module/patch/operator.py#L108-L113

Reply:
Well I wrote it and I'm a bit ashamed ^^ so maybe we can add this lookup dictionary as a constant somewhere in the code ?

Reply:
haha - yeah it ain't pretty